### PR TITLE
Add tests for `RemoveByPath`

### LINKED ISSUE #479: Add tests for `RemoveByPath`
It seems that there are cases when `RemoveByPath` does not work correctly. Add tests, fix problems.

Also, add it to `Array` and `CompositeTypeInterface`.

### PR DESCRIPTION
**Pull request by seeforschauer**
Closes #479.

### CODE REVIEW

**Comment by seeforschauer** [comment]
A test for `RemoveByPath` is in the `testutil` module to avoid import cycle.

**Comment by AlekSi** [comment]
> A test for RemoveByPath is in the testutil module to avoid import cycle.

Let's move it to `types` and use `assert.Equal` as other tests in that package do

**Comment by seeforschauer** [comment]
updated with `assert.Equal`.

File: internal/types/document.go [hunk 1]
```diff
@@ -321,39 +321,42 @@ func (d *Document) Remove(key string) {
 
 // RemoveByPath removes document by path, doing nothing if the key does not exist.
 func (d *Document) RemoveByPath(keys ...string) {
+	removeByPath(d, keys...)
+}
+
+func removeByPath(v any, keys ...string) {
```
Comment (w84thesun) [inline internal/types/document.go:327]: Should we add comment to this function?

Comment (seeforschauer) [reply internal/types/document.go:327]: it's private, the public one with similar name has.

**Review by AlekSi — APPROVED** [review]

**Pull request merged**